### PR TITLE
feat(useTimeoutFn): add callback args & state isPending

### PR DESCRIPTION
**Pull request by liaoliao666**

### CODE REVIEW

File: packages/shared/useTimeoutFn/index.ts [hunk 1]
```diff
@@ -10,11 +10,12 @@ import { isClient } from '../utils'
  * @param immediate
  */
 export function useTimeoutFn(
-  cb: () => any,
+  cb: (...args: unknown[]) => any,
   interval?: number,
   immediate = true,
 ) {
   const isActive = ref(false)
+  const isPending = ref(false)
```
Comment (antfu) [inline packages/shared/useTimeoutFn/index.ts:18]: What is this for?

File: packages/shared/useTimeoutFn/index.ts [hunk 2]
```diff
@@ -27,15 +28,19 @@ export function useTimeoutFn(
...
-  function start() {
+  function start(...args: unknown[]) {
     clear()
     isActive.value = true
+    isPending.value = true
     timer = setTimeout(() => {
+      isPending.value = false
       timer = null
-      cb()
+      // eslint-disable-next-line
```
Comment (antfu) [inline packages/shared/useTimeoutFn/index.ts:42]: Can you make this more specific by only disabling the rule that throws?

File: packages/shared/useTimeoutFn/index.ts (hunk 1 above)
Comment (liaoliao666) [reply packages/shared/useTimeoutFn/index.ts:18]: sometimes i want to know it  whether in pending and i think the isActive is useless. maybe u could intead isPending of isActive
for example
```vue
<template>
  <button :disabled="isPending">
    <slot />
    <Spinner v-if="isPending" />
  </button>
</template>
```

File: packages/shared/useTimeoutFn/index.ts (hunk 2 above)
Comment (liaoliao666) [reply packages/shared/useTimeoutFn/index.ts]: okay

File: packages/shared/useTimeoutFn/index.ts (hunk 1 above)
Comment (antfu) [reply packages/shared/useTimeoutFn/index.ts:18]: I see, but looks like you forget to export it? Yeah, let's remove `isActive` in favor of `isPending`.

**Review by antfu — APPROVED** [review]

**Pull request merged**